src/tile.cpp

```diff
@@ -628,15 +628,15 @@ ReturnValue Tile::queryAdd(int32_t, const Thing& thing, uint32_t, uint32_t flags
 			}
 
 			if (const auto items = getItemList()) {
-				for (const Item* item : *items) {
+				for (const auto& item : *items) {
 					const ItemType& iiType = Item::items[item->getID()];
 					if (iiType.blockSolid && (!iiType.moveable || item->hasAttribute(ITEM_ATTRIBUTE_UNIQUEID))) {
 						return RETURNVALUE_NOTPOSSIBLE;
 					}
 				}
 			}
 		}
-	} else if (const Item* item = thing.getItem()) {
+	} else if (auto item = thing->getItem()) {
```

Comment:
Could this be `const auto`
```suggestion
	} else if (const auto item = thing->getItem()) {
```